### PR TITLE
Schema/seeds

### PR DESCRIPTION
**Pull request by ElectricMolasses**
#10 
- Imported Gorm
- Created Migrations
- Created minimal number of seeds to test FK's
- VendorOrders needs restructuring
- Functions to create entries have been baked into the seed package, will likely want to move them out into their own package and have seeds import it, to avoid unnecessary export of Seed() in prod.
- Currently using hard coded dummy logins that only work locally.

### CODE REVIEW

File: back-end/db/migrations/main.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	"../schema"
+	"../seeds"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	_ "github.com/lib/pq"
+)
+
+func main() {
+	db, err := gorm.Open("postgres", "host=127.0.0.1 port=5432 user=sap dbname=postgres password='password' sslmode=disable")
```
Comment (andydlindsay) [inline back-end/db/migrations/main.go]: Does Go have the equivalent of `dotenv`?

File: back-end/db/schema/schema.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+// VendorOrders Schema
+// This needs to be restrucuted, slight improvement
+// over what's in the ERD but it does not fulfill
+// requirements for the app.
+type VendorOrders struct {
+	gorm.Model
+	User     Users   `gorm:"NOT NULL"`
+	UserID   uint    `gorm:"NOT NULL"`
```
Comment (andydlindsay) [inline back-end/db/schema/schema.go:99]: For this ORM, do we need to specify both that it relates to the User table and that it has a UserID? (Genuine question)

File: back-end/db/seeds/seeds.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	db.Create(&orderItem)
+
+	vendorOrder := createVendorOrder(user, vendor, "Need more fake Bruschetta.")
+	db.Create(&vendorOrder)
+
+	address := createAddress(customer, "Bob", "Ross", "1440 Painter Ave.", "PaintScape", "12235658747", "Home", true)
+	db.Create(&address)
+}
+
+func createUser(email, role, firstName, lastName string) schema.Users {
```
Comment (andydlindsay) [inline back-end/db/seeds/seeds.go:38]: Does `string` indicate that all of the arguments are strings? Or just `lastName`?

File: back-end/db/migrations/main.go (hunk 1 above)
Comment (Rhombert) [reply back-end/db/migrations/main.go]: It does, I'll get that changed over and then update the request.

File: back-end/db/seeds/seeds.go (hunk 3 above)
Comment (Rhombert) [reply back-end/db/seeds/seeds.go:38]: Everything preceding it, until it hits another type, so yes, it does.  As an example
(number1, number2 int, word1, word2, word3 string) is a better example of how it works.

File: back-end/db/schema/schema.go (hunk 2 above)
Comment (Rhombert) [reply back-end/db/schema/schema.go:99]: I believe so, I just tested to be safe and it didn't actually create the FK's if I remove the UserID, and it fails to associate the FK's if I remove User.  So one is for the ORM's connection, and the other is for the actual database?  Everything I see in the docs points to me using both as well.

**Comment by ElectricMolasses** [comment]
I've added a new config module to handle additions/removals from the yml without having to manually make those changes every time when we read settings from said file, and included viper to parse the yml for the config.  That aside, everything else is just modifications to the files that already existed as per comments above.

File: back-end/config/config.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+	// the db live for faster testing.
+	// viper.WatchConfig()
+	// It also has an event handler for changes
+	// viper.OnConfigChange(func)
+
+	if err := viper.ReadInConfig(); err != nil {
+		fmt.Printf("Error reading config file, %s", err)
+	}
+
+	dbVars := viper.Sub("database")
```
Comment (Robert-Beckett) [inline back-end/config/config.go]: sub just accesses a sublevel like a node in the tree, and all my config settings are stored under database.

**Pull request merged**